app/src/components/Overlay.vue

```diff
@@ -0,0 +1,24 @@
+<template>
+  <div class="Overlay">
+    <slot></slot>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Overlay"
+};
+</script>
+
+<style lang="scss">
+.Overlay {
+  position: fixed;
+  z-index: 1;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
```

Comment:
Using percentage here will make the overlay have the size of the whole page. Surely, what we want is to fill the screen, not the whole page.

```suggestion
  width: 100vw;
  height: 100vh;
```

Reply:
Also, to make sure that the background cannot be scrolled while the overlay is active. We must disable scrolling on the `body` element.

```css
body {
  overflow-y: hidden;
}
```

To do this only when the overlay is rendered, we can use the `mounted` and `beforeUnmount` lifecycle hooks.

```javascript
export default {
  mounted() {
    document.body.style.overflowY = 'hidden';
  },
  beforeUnmount() {
    document.body.style.overflowY = '';
  }
}
```

Reply:
The overflow trick does not work on a mobile device. I'll use your suggestions for now, and we should manage the mobile part in another issue.